xarray/tests/test_datatree.py

```diff
@@ -166,62 +167,63 @@ def test_assign_when_already_child_with_variables_name(self):
             dt.ds = new_ds
 
 
-class TestGet:
-    ...
+class TestGet: ...
 
 
 class TestGetItem:
     def test_getitem_node(self):
-        folder1 = DataTree(name="folder1")
-        results = DataTree(name="results", parent=folder1)
-        highres = DataTree(name="highres", parent=results)
+        folder1: DataTree = DataTree(name="folder1")
+        results: DataTree = DataTree(name="results", parent=folder1)
+        highres: DataTree = DataTree(name="highres", parent=results)
         assert folder1["results"] is results
         assert folder1["results/highres"] is highres
 
     def test_getitem_self(self):
-        dt = DataTree()
+        dt: DataTree = DataTree()
         assert dt["."] is dt
 
     def test_getitem_single_data_variable(self):
         data = xr.Dataset({"temp": [0, 50]})
-        results = DataTree(name="results", data=data)
+        results: DataTree = DataTree(name="results", data=data)
         xrt.assert_identical(results["temp"], data["temp"])
 
     def test_getitem_single_data_variable_from_node(self):
         data = xr.Dataset({"temp": [0, 50]})
-        folder1 = DataTree(name="folder1")
-        results = DataTree(name="results", parent=folder1)
+        folder1: DataTree = DataTree(name="folder1")
+        results: DataTree = DataTree(name="results", parent=folder1)
         DataTree(name="highres", parent=results, data=data)
         xrt.assert_identical(folder1["results/highres/temp"], data["temp"])
 
     def test_getitem_nonexistent_node(self):
-        folder1 = DataTree(name="folder1")
+        folder1: DataTree = DataTree(name="folder1")
         DataTree(name="results", parent=folder1)
         with pytest.raises(KeyError):
             folder1["results/highres"]
 
     def test_getitem_nonexistent_variable(self):
         data = xr.Dataset({"temp": [0, 50]})
-        results = DataTree(name="results", data=data)
+        results: DataTree = DataTree(name="results", data=data)
         with pytest.raises(KeyError):
             results["pressure"]
 
     @pytest.mark.xfail(reason="Should be deprecated in favour of .subset")
     def test_getitem_multiple_data_variables(self):
         data = xr.Dataset({"temp": [0, 50], "p": [5, 8, 7]})
-        results = DataTree(name="results", data=data)
-        xrt.assert_identical(results[["temp", "p"]], data[["temp", "p"]])
+        results: DataTree = DataTree(name="results", data=data)
+        xrt.assert_identical(results[["temp", "p"]], data[["temp", "p"]])  # type: ignore[index]
```

Comment:
I think this is reasonable - this and `test_getitem_dict_like_selection_access_to_dataset` are using `Iterable` types (a list and a dictionary), which currently would raise exceptions (and the tests are marked as `xfail`).  Other options I considered included writing tests that would make sure the expected assertions were raised, but it feels like the actual desired behaviour is TBD, and that the best thing to do was to leave things as they were.